### PR TITLE
ARROW-16375: [R][CI] Pin test-r-devdocs on Windows to R 4.1

### PR DESCRIPTION
**Pull request by thisisnic**

### CODE REVIEW

**Review by wjones127 — APPROVED** [review]

File: dev/tasks/r/github.devdocs.yml [hunk 1]
```diff
@@ -32,6 +32,8 @@ jobs:
       {{ macros.github_checkout_arrow()|indent }}
 
       - uses: r-lib/actions/setup-r@v1
+        with:
+          r-version: '4.1'
```
Comment (nealrichardson) [inline dev/tasks/r/github.devdocs.yml:36]: Should we leave a comment here explaining why we're pinning a version? Link to a jira to change that?

**Review by nealrichardson — APPROVED** [review]
One suggestion about adding a comment (my favorite request) but the fix LGTM

File: dev/tasks/r/github.devdocs.yml [hunk 2]
```diff
@@ -32,6 +32,8 @@ jobs:
       {{ macros.github_checkout_arrow()|indent }}
 
       - uses: r-lib/actions/setup-r@v1
```
[the remark this answers is not in the document]
Comment (thisisnic) [reply dev/tasks/r/github.devdocs.yml]: Yep!

File: dev/tasks/r/github.devdocs.yml [hunk 3]
```diff
@@ -31,7 +31,12 @@ jobs:
...
       {{ macros.github_checkout_arrow()|indent }}
 
-      - uses: r-lib/actions/setup-r@v1
+      - uses: r-lib/actions/setup-r@v2
+        with:
+          # temp workaround the fact that R 4.2 requires UCRT which isn't
+          # currently part of this build.
+          # remove after https://issues.apache.org/jira/browse/ARROW-16376
+          r-version: '4.1'
       - uses: r-lib/actions/setup-pandoc@v1
```
Comment (nealrichardson) [inline dev/tasks/r/github.devdocs.yml:40]: Update them all?
```suggestion
      - uses: r-lib/actions/setup-pandoc@v2
```

**Pull request closed without merging**